feflow/utils/hybrid_topology.py

```diff
@@ -56,23 +124,2437 @@ def __init__(self,
+            # atom
+            new_system_res = new_system_atom.residue
+
+            # Next, we have to enumerate the other atoms in that residue to
+            # find mapped atoms
+            new_system_atom_set = {atom.index for atom in new_system_res.atoms}
+
+            # Now, we find the subset of atoms that are mapped. These must be
+            # in the "core" category, since they are mapped and part of a
+            # changing residue
+            mapped_new_atom_indices = core_atoms_new_indices.intersection(
+                new_system_atom_set)
+
+            # Now get the old indices of the above atoms so that we can find
+            # the appropriate residue in the old system for this we can use the
+            # new to old atom map
+            mapped_old_atom_indices = [self._new_to_old_map[atom_idx] for
+                                       atom_idx in mapped_new_atom_indices]
+
+            # We can just take the first one--they all have the same residue
+            first_mapped_old_atom_index = mapped_old_atom_indices[0]
+
+            # Get the atom object corresponding to this index from the hybrid
+            # (which is a deepcopy of the old)
+            mapped_hybrid_system_atom = hybrid_topology.atom(
+                first_mapped_old_atom_index)
+
+            # Get the residue that is relevant to this atom
+            mapped_residue = mapped_hybrid_system_atom.residue
+
+            # Add the atom using the mapped residue
+            added_atoms[new_particle_hybrid_idx] = hybrid_topology.add_atom(
+                new_system_atom.name,
+                new_system_atom.element,
+                mapped_residue)
+
+        # Now loop through the bonds in the new system, and if the bond
+        # contains a unique new atom, then add it to the hybrid topology
+        for (atom1, atom2) in new_top.bonds:
+            at1_hybrid_idx = self._new_to_hybrid_map[atom1.index]
+            at2_hybrid_idx = self._new_to_hybrid_map[atom2.index]
+
+            # If at least one atom is in the unique new class, we need to add
+            # it to the hybrid system
+            at1_uniq = at1_hybrid_idx in self._atom_classes['unique_new_atoms']
+            at2_uniq = at2_hybrid_idx in self._atom_classes['unique_new_atoms']
+            if at1_uniq or at2_uniq:
+                if at1_uniq:
+                    atom1_to_bond = added_atoms[at1_hybrid_idx]
+                else:
+                    old_idx = self._hybrid_to_old_map[at1_hybrid_idx]
+                    atom1_to_bond = hybrid_topology.atom(old_idx)
+
+                if at2_uniq:
+                    atom2_to_bond = added_atoms[at2_hybrid_idx]
+                else:
+                    old_idx = self._hybrid_to_old_map[at2_hybrid_idx]
+                    atom2_to_bond = hybrid_topology.atom(old_idx)
+
+                hybrid_topology.add_bond(atom1_to_bond, atom2_to_bond)
+
+        return hybrid_topology
+
+    def _create_hybrid_topology(self):
```

Comment:
This is new code I added a couple of versions back for openfe - I think the tests for it haven't been migrated either.

(It shouldn't affected anything in perses because it used to be a dead branch, just now it works).